### PR TITLE
Show hash on gateway

### LINKED ISSUE #5724: Gateways should always display "Index of /ipfs/HASH", instead of the URL.
#### Version information:
go-ipfs version: 0.4.18-
Repo version: 7
System version: amd64/linux
Golang version: go1.11.1

#### Type:
Bug/Feature?

#### Description:
Currently when viewing a directory on a gateway, it will display "Index of <url>", which can be anything from `/` to `/ipns/domain.com` to `/ipfs/HASH`.

I think it should instead ALWAYS display the ipfs hash that's being shown, as this is both more consistent, more useful in case of screenshots (have fun trying to access a dnslink in 20 years), and makes better use of space since the URL will already be displayed in the browser's address bar.

There's also the option of displaying both the IPFS hash, and the URL, which is probably the best option if it's not difficult to implement.

### PR DESCRIPTION
**Pull request by jackloughran**
@Stebalien this will show the hash if the request doesn't have ipfs in the url:
image
image

fixes #5724

### CODE REVIEW

**Comment by jackloughran** [comment]
@Stebalien yes: https://github.com/ipfs/dir-index-html/pull/21. sorry, ran out of time yesterday

File: core/corehttp/gateway_handler.go [hunk 1]
```diff
@@ -347,11 +347,17 @@ func (i *gatewayHandler) getOrHeadHandler(ctx context.Context, w http.ResponseWr
 		}
 	}
 
+	var hash string
+	if !strings.Contains(originalUrlPath, "ipfs") {
```
Comment (Stebalien) [inline core/corehttp/gateway_handler.go]: We should use HasPrefix with `ipfsPathPrefix`. This won't catch IPNS links with the word `ipfs` somewhere in the path.

**Comment by Stebalien** [comment]
(note: I've updated dir-index-html and rebased this)

**Comment by Stebalien** [comment]
@kevina, @magik6k, and/or @swedneck I'd like someone else to sign off on this as both @jackloughran and I participated a bit too much in the design to be neutral third parties.

**Comment by swedneck** [comment]
I can't judge the code, but the screenshots look great.

**Comment by jackloughran** [comment]
@Stebalien thanks for making that change. i got sidetracked with the holidays. looks like this is good to go, but i'd be happy to do anything else over the weekend.

**Pull request merged**